app_flutter/lib/index_page.dart

```diff
@@ -6,147 +6,83 @@ import 'package:flutter/material.dart';
-    super.dispose();
-  }
-}
-
-class Index extends StatelessWidget {
-  const Index({
-    Key key,
-    this.scaffoldKey,
   }) : super(key: key);
 
-  final GlobalKey<ScaffoldState> scaffoldKey;
+  static const String routeName = '/';
 
   static const Widget separator = SizedBox(height: 24.0);
 
   @override
   Widget build(BuildContext context) {
     final ThemeData theme = Theme.of(context);
-    return Consumer<IndexState>(
-      builder: (BuildContext context, IndexState indexState, Widget child) => Scaffold(
-        key: scaffoldKey,
+    final IndexState indexState = Provider.of<IndexState>(context);
+    return AnimatedBuilder(
+      animation: indexState,
+      builder: (BuildContext context, Widget child) => Scaffold(
         appBar: AppBar(
+          // TODO(ianh): factor out this code so other pages can use it too without code duplication
           title: const Text('Cocoon'),
           actions: <Widget>[
             SignInButton(
-              authService: indexState.authService,
               colorBrightness: theme.appBarTheme.brightness ?? theme.primaryColorBrightness,
             ),
           ],
         ),
-        body: Center(
-          child: IntrinsicWidth(
-            stepWidth: 80.0,
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: <Widget>[
-                separator,
-                Text('Select a dashboard', style: Theme.of(context).textTheme.headline4, textAlign: TextAlign.center),
-                separator,
-                RaisedButton.icon(
-                  icon: Icon(Icons.build),
-                  label: const Text('BUILD'),
-                  onPressed: () => Navigator.pushReplacementNamed(context, '/build'),
-                ),
-                separator,
-                RaisedButton.icon(
-                  icon: Icon(Icons.show_chart),
-                  label: const Text('BENCHMARKS'),
-                  onPressed: () => launch('/benchmarks.html'),
-                ),
-                separator,
-                RaisedButton.icon(
-                  icon: Icon(Icons.show_chart),
-                  label: const Text('BENCHMARKS ON SKIA PERF'),
-                  onPressed: () => launch('https://flutter-perf.skia.org/'),
-                ),
-                separator,
-                RaisedButton.icon(
-                  icon: Icon(Icons.info_outline),
-                  label: const Text('REPOSITORY'),
-                  onPressed: () => launch('/repository.html'),
-                ),
-                separator,
-                const Divider(thickness: 2.0),
-                separator,
-                RaisedButton.icon(
-                  icon: Icon(Icons.android),
-                  label: const Text('INFRA AGENTS'),
-                  onPressed: () => Navigator.pushReplacementNamed(context, '/agents'),
-                ),
-              ],
+        body: ErrorBrookWatcher(
```

Comment:
I dunno man, it listened for errors before, I just reproduced what there was before. :-)

I'm happy to remove this if you don't want it. (I can also remove the `errors` from IndexState since they have no effect either).